### PR TITLE
Add start_listening command

### PR DESCRIPTION
**Pull request by balloob**
- Add basis for dealing with commands
- Add `start_listening` command
- Add `--mock-driver` flag to `cli/server.ts` for easy testing
- Rename `lib/{express,server}.ts`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,15 +40,37 @@ ts-node cli/client.ts --dump
...
+```ts
+interface {
+  type: "version";
+  driverVersion: string;
+  serverVersion: string;
+  homeId: number;
+}
+```
+
+To start receive the state and get events, the client needs to send the `start_listening command.
```
Comment (marcelveldt) [inline README.md:54]: small typo 'start_listening'

File: README.md [hunk 2]
```diff
@@ -38,15 +40,37 @@ ts-node cli/client.ts --dump
...
+  command: "start_listening";
+}
+```
+
+The server will respond with the current state and start sending events.
+
+```ts
+interface {
+  type: "result";
+  messageID: string; // maps the `start_listening` command
```
Comment (marcelveldt) [inline README.md:68]: just 'id' or 'messageId' ?

Comment (balloob) [reply README.md:68]: I know used messageID. Prefer messageId ?

File: README.md (hunk 1 above)
Comment (balloob) [reply README.md:54]:
```suggestion
To start receive the state and get events, the client needs to send the `start_listening` command.
```

**Pull request merged**